lib/config.ts

```diff
@@ -1,3 +1,5 @@
+export interface HttpProxyAgent { constructor(opts: Object): HttpProxyAgent; }
```

Comment:
I agree, we should remove the type in the config. `HttpProxyAgent` does not have an ambient typings; however, we should not have to define the interface for the constructor.